qcodes/instrument_drivers/tektronix/DPO7200xx.py

```diff
@@ -0,0 +1,704 @@
+
+        self.add_parameter(
+            "scale",
+            get_cmd=f"{self._identifier}:SCA?",
+            set_cmd=f"{self._identifier}:SCA {{}}",
+            get_parser=float,
+            unit="V/div"
+        )
+
+        self.add_parameter(
+            "offset",
+            get_cmd=f"{self._identifier}:OFFS?",
+            set_cmd=f"{self._identifier}:OFFS {{}}",
+            get_parser=float,
+            unit="V"
+        )
+
+        self.add_parameter(
+            "position",
+            get_cmd=f"{self._identifier}:POS?",
+            set_cmd=f"{self._identifier}:POS {{}}",
+            get_parser=float,
+            unit="V"
+        )
+
+        self.add_parameter(
+            "termination",
+            get_cmd=f"{self._identifier}:TER?",
+            set_cmd=f"{self._identifier}:TER {{}}",
+            vals=Enum(50, 1E6),
+            get_parser=float,
+            unit="Ohm"
+        )
+
+        self.add_parameter(
+            "analog_to_digital_threshold",
+            get_cmd=f"{self._identifier}:THRESH?",
+            set_cmd=f"{self._identifier}:THRESH {{}}",
+            get_parser=float,
+            unit="V",
+        )
+
+        self.add_parameter(
+            "termination_voltage",
+            get_cmd=f"{self._identifier}:VTERm:BIAS?",
+            set_cmd=f"{self._identifier}:VTERm:BIAS {{}}",
+            get_parser=float,
+            unit="V"
+        )
+
+    def set_trace_length(self, value: int) -> None:
+        """
+        Set the trace length when retrieving data
+        through the 'waveform' interface
+
+        Args:
+            The requested number of samples in the trace
+
+        Example:
+
+        """
+        if self.root_instrument.horizontal.record_length() < value:
+            raise ValueError(
+                "Cannot set a trace length which is larger than "
+                "the record length. Please switch to manual mode "
+                "and adjust the record length first"
+            )
+
+        self.root_instrument.data.start_index(1)
+        self.root_instrument.data.stop_index(value)
+
+    def set_trace_time(self, value: float) -> None:
+        """
+        Args:
+            The time over which a trace is desired.
+        """
+        sample_rate = self.root_instrument.horizontal.sample_rate()
+        required_sample_count = int(sample_rate * value)
+        self.set_trace_length(required_sample_count)
+
+
+class TektronixDPOHorizontal(InstrumentChannel):
+    """
+    This module controls the horizontal axis of the scope
+    """
+
+    def __init__(
+            self,
+            parent: Union[Instrument, InstrumentChannel],
+            name: str
+    ) -> None:
+
+        super().__init__(parent, name)
+
+        self.add_parameter(
+            "mode",
+            get_cmd="HORizontal:MODE?",
+            set_cmd="HORizontal:MODE {}",
+            vals=Enum("auto", "constant", "manual"),
+            get_parser=str.lower,
+            docstring="""
```

Comment:
do these long docstrings render nicely in jupyter, meaning with correct spacing and indentation? if not, then i'd suggest to wrap them in a call to [`textwrap.dedent`](https://docs.python.org/3.6/library/textwrap.html#textwrap.dedent) (i think it's done that way in some other driver already).